### ISSUE TITLE
cycling of `condition case` warnings when using `use-package`

### ISSUE THREAD

**Issue by 00krishna** [jwiegley/use-package#557]
Hello John. 

Say, I ran into some odd behavior when using `Spacemacs` and `Orgmode` and the issue seems related to use-package. I was hoping you might know what is causing this. 

When I try and invoke a new `org-capture` template I get this error message below. But the odd thing is that it cycles between warning349 and warning 391. Not sure why that is. 
```
condition-case: Symbol’s value as variable is void: use-package--warning349
condition-case: Symbol’s value as variable is void: use-package--warning931
condition-case: Symbol’s value as variable is void: use-package--warning349
condition-case: Symbol’s value as variable is void: use-package--warning931
condition-case: Symbol’s value as variable is void: use-package--warning349
```
Emacs boots just fine--so there is no backtrace. I also tried to delete and reload the use-package package. But no luck. Not sure what to give you so you can replicate the issue. 

The only changes I made involved trying to get org-protocol working. But the only only changes I made to my config had to do with the letter used for the specific org-capture templates. So otherwise should be good. 

Please let me know if you have any ideas. Thanks.

**Comment by jwiegley** [comment]
So, I was wondering if this would happen; specifically, whether condition-case might call its error handler outside the lexical scope of its use.

The fix will be to use `lexical-let` instead. I was avoiding it thinking it might not be necessary, but apparently it is.

You can try this for me locally if you'd be so kind: On lines 929 and 949 of `use-package-core.el`, just change the word `let` there to `lexical-let`. Be sure to re-byte-compile your `use-package-core.el` afterward. Then restart Emacs.

**Comment by 00krishna** [comment]
Okay cool. I will give it a shot and let you know how it goes. Thanks for the response.

**Comment by 00krishna** [comment]
Hey @jwiegley looks like this change fixed the issue. I did as you said and converted to `lexical-let` on those two lines. Then I recompiled the file and now the org-capture templates work. Haha, this change also fixed an issue I was having with org-protocol as well--due to the connection between org-protocol and org-capture.

Now I just get an error about the the signature--probably because I recompiled the file myself. But just for your FYI, here is the message:

```
Failed to verify signature archive-contents.sig:
Bad signature from 474F05837FBDEF9B GNU ELPA Signing Agent <elpasign@elpa.gnu.org>
Command output:
gpg: Signature made Tue 05 Dec 2017 02:10:02 AM PST using DSA key ID 7FBDEF9B
gpg: BAD signature from "GNU ELPA Signing Agent <elpasign@elpa.gnu.org>" [unknown]
```

Thanks so much for your help.
